lib/spack/spack/test/cmd/env.py

```diff
@@ -3108,17 +3117,67 @@ def test_environment_depfile_makefile(depfile_flags, expected_installs, tmpdir,
     # Do make dry run.
     out = make("-n", "-f", makefile, output=str)
 
-    # Spack install commands are of the form "spack install ... # <spec>",
-    # so we just parse the spec again, for simplicity.
-    specs_that_make_would_install = [
-        Spec(line.split("# ")[1]).name for line in out.splitlines() if line.startswith("spack")
-    ]
+    specs_that_make_would_install = _parse_dry_run_package_installs(out)
 
     # Check that all specs are there (without duplicates)
     assert set(specs_that_make_would_install) == set(expected_installs)
     assert len(specs_that_make_would_install) == len(expected_installs)
 
 
+@pytest.mark.parametrize(
+    "picked_package,expected_installs",
+    [
+        (
+            "dttop",
+            [
+                "dtbuild2",
+                "dtlink2",
+                "dtrun2",
+                "dtbuild1",
+                "dtlink4",
+                "dtlink3",
+                "dtlink1",
+                "dtlink5",
+                "dtbuild3",
+                "dtrun3",
+                "dtrun1",
+                "dttop",
+            ],
+        ),
+        ("dtrun1", ["dtlink5", "dtbuild3", "dtrun3", "dtrun1"]),
+    ],
+)
+def test_depfile_phony_convenience_targets(
+    picked_package, expected_installs, tmpdir, mock_packages
+):
+    """Check if install/{name}-{version}-{hash} targets are created for"""
+    make = Executable("make")
+    with fs.working_dir(str(tmpdir)):
+        with ev.Environment("."):
+            add("dttop")
+            concretize()
+
+        with ev.Environment(".") as e:
+            picked_spec = e.matching_spec(picked_package)
+            env("depfile", "-o", "Makefile", "--make-disable-jobserver")
+
+        # Phony install/* target should install picked package and all its deps
+        specs_that_make_would_install = _parse_dry_run_package_installs(
+            make("-n", picked_spec.format("install/{name}-{version}-{hash}"), output=str)
+        )
+
+        assert set(specs_that_make_would_install) == set(expected_installs)
+        assert len(specs_that_make_would_install) == len(expected_installs)
```

Comment:
Why are we also checking the length here? Do we care if one has more duplicates than the other?

Reply:
yes, there shouldn't be duplicates

Reply:
In that case shouldn't we:
```python
assert set(foo) == set(bar)
assert len(foo) == len(set(foo))
```
instead of:
```python
assert set(foo) == set(bar)
assert len(foo) == len(bar)
```
?

Reply:
sure, that's technically correct, but feels unnecessary given that the packages are listed above.

Reply:
Can we do this small update to enforce the invariant?